src/gui/inputdevices/TouchInputHandler.cpp

```diff
@@ -137,16 +132,25 @@ void TouchInputHandler::zoomStart() {
         zoomControl->setZoomFitMode(false);
     }
 
+    auto center = (this->priLastAbs + this->secLastAbs) / 2.0;
+
     auto* mainWindow = inputContext->getView()->getControl()->getWindow();
+    // use screen pixel coordinates for the zoom center
+    // as relative coordinates depend on the changing zoom level
+    int rx, ry;
+    gdk_window_get_root_coords(gtk_widget_get_window(mainWindow->getWinXournal()), 0, 0, &rx, &ry);
+    center.x -= rx;
+    center.y -= ry;
```

Comment:
Instead of exposing `getWinXournal()`, it might make sense to add a `MainWindow::getScreenPos() -> Point` that calls `Util::toScreenCoords(WindowWidget, Point) -> Point`.

```suggestion
    // use screen pixel coordinates for the zoom center
    // as relative coordinates depend on the changing zoom level
    utl::Point<double> windowTopLeft = mainWindow->getScreenPos();
    center.x -= windowTopLeft.x;
    center.y -= windowTopLeft.y;
```